designs/tripleo/osc-tripleo.md

```diff
@@ -140,8 +149,34 @@ OSC requires a mounted volume for persistence of data such us the OSC H2 databas
 The commands executed within the OSC container will done with the *root* user as defined here `user: root`.
 > Assumption: The THT community does not have reservations with respect to using the root user within the container. Other OOO container services use the same approach, for instance [nova](https://github.com/openstack/tripleo-heat-templates/blob/master/docker/services/nova-api.yaml) or [aodh](https://github.com/openstack/tripleo-heat-templates/blob/master/docker/services/aodh-api.yaml). The root user is not a strong requirement for OSC but understanding the full impact of changing that we will require further investigation. 
 
-#### OSC Service Ports  
-TODO: How are the ports used by the service provided through the THT?
+### OSC Service Ports
+The OSC service makes use of two ports to expose its functionality: the OSC APIs use the port `8090` and the UI `443` within its container. Those ports must be mapped to the host ports dedicated to OSC. When running the OSC container on Docker these bindings are provided through the Docker run parameter `-p`, for details see [Docker networking](https://docs.docker.com/config/containers/container-networking/). For a OOO deployment, the following configuration is assumed to provide the same effect:
+```yaml
+tripleo.osc.firewall_rules:
+              'osc-ui':
+                dport:
+                  -  {get_param: [EndpointMap, OSCUIPublic, port]}
+                  - 443
+              'osc-api':
+                dport:
+                  -  {get_param: [EndpointMap, OSCAPIPublic, port]}
+                  - 8090
+```
+> **Assumption:** The Docker run param `-p` effectively creates a firewall rule which maps the container port to a port on the Docker host. Unfortunately, the [THT Containers documentation](http://tripleo.org/install/containers_deployment/architecture.html) does not make any reference to port binding. Looking at other existing THT services the closest way to configure this is with an entry similar to above but this semantics and syntax must be validated with a THT/RH engineer.
+
+The value for the endpoint ports `OSCUIPublic` and `OSCAPIPublic` must be defined in the file `network/endpoints/endpoint_data.yaml` for OSC as shown below:
```

Comment:
I assume these are the ports on the container hosting machine?

Reply:
That is right